src/components/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,160 @@
+  /** Populates the placeholder of the input */
+  placeholder?: string;
+  /** Populates the value of the input */
+  value?: string | number;
+  /** className you can add in addition to 'input' */
+  className?: string;
+  /** ID that other components can cross reference for accessibility purposes */
+  id?: string;
+  /** Helper for modifiers array; adds 'errored' styling */
+  errored?: boolean;
+  /** Adds the 'disabled' prop to the input when true */
+  disabled?: boolean;
+  /** Modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
+  modifiers?: string[];
+  /** The action to perform on the `<input>`'s onChange function  */
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+const TextInput = React.forwardRef<HTMLInputElement, InputProps>(
+  (props, ref?) => {
+    const {
+      type = TextInputTypes.text,
+      required,
+      labelText,
+      showLabel = true,
+      helperText,
+      errorText,
+      placeholder,
+      value,
+      className,
+      disabled,
+      errored,
+      onChange,
+    } = props;
+
+    const attributes = props.attributes || {};
+    const modifiers = props.modifiers ? props.modifiers : [];
+    const id = props.id || generateUUID();
+
+    if (!showLabel) attributes["aria-label"] = labelText;
+    if (helperText) attributes["aria-describedby"] = helperText;
+
+    if (errored) {
+      modifiers.push("error");
+    }
+    if (type === TextInputTypes.textarea) {
+      modifiers.push("textarea");
+    }
+
+    const errorOutput = errorText
+      ? errorText
+      : "There is an error related to this field.";
+
+    let footnote;
+    if (errored) {
+      footnote = errorOutput;
+    } else {
+      footnote = helperText;
+    }
+
+    if (
+      type === TextInputTypes.tel ||
+      type === TextInputTypes.url ||
+      type === TextInputTypes.email
+    ) {
+      const example = TextInputFormats[type];
+      footnote = (
+        <>
+          Ex: {example}
+          <br />
+          {footnote}
+        </>
+      );
+    }
+
+    const textField = (
+      <>
+        <input
+          id={id}
+          className={bem("inputfield", modifiers, "", [className])}
+          type={type}
+          value={value}
+          aria-required={required}
+          aria-hidden={type === TextInputTypes.hidden}
+          disabled={disabled}
+          required={required}
+          placeholder={placeholder}
+          onChange={onChange}
+          ref={ref}
+          {...attributes}
+        />
+      </>
+    );
+    const textareaField = (
+      <>
+        <textarea
+          id={id}
+          className={bem("inputfield", modifiers, "", [className])}
+          aria-required={required}
+          aria-hidden={type === TextInputTypes.hidden}
+          disabled={disabled}
+          required={required}
+          placeholder={placeholder}
+          // onChange={onChange}
+          // ref={ref}
```

Comment:
Curious why these are commented-out?

Reply:
@kaplanmichael 
I commented out these two lines of code because they were throwing an error during development, but, at the time, I was not sure why that was the case.  I then decided to ignore those two lines and moved on.  However, I now see that the errors come up are due to what is being declared on line 41.  The forwardRef is referencing `HTMLInputElement`, but the element where I commented the code is `HTMLTextAreaElement`.  How can I assign that dynamically?

Reply:
I see that you've commented out the text area which is fine for now.  Happy to help you troubleshoot that--even if we spin that off into another ticket.  One thing though:  I recommend you also comment textarea out of the textinputtypes file as well, otherwise it will still be selectable in the dropdown: 
![image](https://user-images.githubusercontent.com/6432596/116151966-50ce8280-a6b3-11eb-8c09-0492f2d40863.png)

Reply:
Good catch.  I commented out `textarea`.

Reply:
I also created a new Jira ticket so we can address `textarea` in the future.
https://jira.nypl.org/browse/DSD-244